### PR TITLE
General stuff

### PR DESCRIPTION
**Pull request by h-2**
This doesn't contain any of the new IO code, yet, but it contains the changes necessary in SeqAn3 when removing the existing IO code and it also contains some utility stuff that I think would be useful in general.

### CODE REVIEW

File: include/seqan3/alphabet/detail/debug_stream_alphabet.hpp [hunk 1]
```diff
@@ -30,7 +29,7 @@ namespace seqan3
 template <typename char_t, alphabet alphabet_t>
 inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, alphabet_t && l)
 //!\cond
-    requires (!output_stream_over<std::basic_ostream<char_t>, alphabet_t>)
+    requires (!std::integral<std::remove_cvref_t<alphabet_t>>)
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/debug_stream_alphabet.hpp:32]: is this really equivalent?

File: include/seqan3/utility/concept/one_of.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+/*!\interface   seqan3::one_of <>
+ * \ingroup utility_concept
+ * \tparam t    The query type to compare.
+ * \tparam ts   The reference types.
+ * \brief       Checks whether the list of reference types contains the query type.
+ */
+//!\cond
+template <typename t, typename ... ts>
+concept one_of = (std::same_as<t, ts> || ...);
```
Comment (smehringer) [inline include/seqan3/utility/concept/one_of.hpp:29]: This rather looks like it should be part of the `type_traits/list` and `type_traits/pack` utility.

File: include/seqan3/utility/tag.hpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+namespace seqan3
+{
+
+/*!\brief The type of seqan3::vtag.
+ * \tparam more_vs Default "specialisation" for 0 arguments.
+ * \see seqan3::vtag
+ */
+template <auto ... more_vs>
+struct vtag_t
```
Comment (smehringer) [inline include/seqan3/utility/tag.hpp:29]: what the `v` in `vtag` stand for

File: include/seqan3/alphabet/detail/debug_stream_alphabet.hpp (hunk 1 above)
Comment (h-2) [reply include/seqan3/alphabet/detail/debug_stream_alphabet.hpp:32]: It is not equivalent, but it does what it is supposed to at that point ^^, i.e. prevent char, int, ... from taking this overload.

File: include/seqan3/utility/concept/one_of.hpp (hunk 2 above)
Comment (h-2) [reply include/seqan3/utility/concept/one_of.hpp:29]: Hm, but it is not for working with lists or packs. 

It does this:

```cpp
void foobar(one_of<std::string_view, std::string> auto & s)
{
  ....
}
```

File: include/seqan3/utility/tag.hpp (hunk 3 above)
Comment (h-2) [reply include/seqan3/utility/tag.hpp:29]: **v**alue tag and **t**ype tag.

**Pull request closed without merging**